**Instructions  
It is assumed that the mean systolic blood pressure is μ = 120 mm Hg. In the Honolulu Heart Study, a sample of n = 100 people had an average systolic blood pressure of 130.1 mm Hg with a standard deviation of 21.21 mm Hg. Is the group significantly different (with respect to systolic blood pressure!) from the regular population?**

** 
Write down all the steps followed for setting up the test.  
Calculate the test statistic by hand and also code it in Python. It should be 4.76190. What decision can you make based on this calculated value?**

**1. Set up the hypothesis test**

In [25]:
import numpy as np
import pandas as pd
import statistics as stats
import scipy.stats as stats
import math
import matplotlib.pyplot as plt
import seaborn as sns

%matplotlib inline

In [12]:
HH_Study_mean=120
sample_mean=130.1
st_dev=21.21


In [13]:
##z distribution is a standard normal distribution as our sample size is bigger than 30
z = (sample_mean-HH_Study_mean)/st_dev*math.sqrt(100)
print("The z score of our sample is: {:.2f}".format(z))

The z score of our sample is: 4.76


In [15]:
zc = stats.t.ppf(0.95, df=99) # 0.95 is the confidence level.
print("The critical value corresponding to a 0.95 area of a normal distribution is: {:.2f}".format(zc))

The critical value corresponding to a 0.95 area of a normal distribution is: 1.66


**As z=4.76 is significantly > than zc = 1.66 -> We reject the null hypothesis. Our group significantly different (with respect to systolic blood pressure!) from the regular population**




In [18]:
# degrees of freedom n-1, you use student t when n<30 or when we do not have a population standard deviation

**Optional**

In a packing plant, a machine packs cartons with jars. It is supposed that a new machine will pack faster on the average than the machine currently used. To test that hypothesis, the times it takes each machine to pack ten cartons are recorded. The results, in seconds, are shown in the tables in the file Data/machine.txt. Assume that there is sufficient evidence to conduct the t test, does the data provide sufficient evidence to show if one machine is better than the other.
Hint: use two sample t-test  

t= (sample_mean(x1) - sample_mean(x2)) / sqrt(square(s1)/n1 + square(s2)/n2 )  



In [36]:
new_machine=[42.1, 41, 41.3, 41.8, 42.4, 42.8, 43.2, 42.3, 41.8, 42.7]
old_machine=[42.7,  43.6, 43.8, 43.3, 42.5,43.5, 43.1, 41.7, 44, 44.1 ]

In [37]:
import statistics as stats
new_machine_mean=stats.mean(new_machine)
new_machine_mean

42.14

In [38]:
old_machine_mean=stats.mean(old_mashine)
old_machine_mean

43.230000000000004

In [39]:
new_machine_stdev=stats.stdev(new_machine)
new_machine_stdev

0.6834552736727638

In [40]:
old_machine_stdev=stats.stdev(old_machine)
old_machine_stdev

0.7498888806572158

In [55]:
t=(new_machine_mean-old_machine_mean) / (math.sqrt(math.sqrt(new_machine_stdev))/10 +math.sqrt(math.sqrt(old_machine_stdev))/10)
t

-5.924530365791009

In [58]:
import scipy.stats as stats
zc = stats.t.ppf(0.95, df=9)
zc

1.8331129326536335

 t=-5.9  >  zc = 1.83